components/board/fake/board.go

```diff
@@ -284,6 +285,7 @@ func (a *Analog) reset(pin string) {
 }
 
 func (a *Analog) Read(ctx context.Context, extra map[string]interface{}) (int, error) {
+	a.Set(rand.Intn(100))
 	a.Mu.RLock()
 	defer a.Mu.RUnlock()
```

Comment:
this was the locking issue actually

Reply:
This makes me nervous: by using the struct's state without first locking the mutex, we have the potential for race conditions. I haven't thought it through all the way, but if you just lock things at the very beginning, you don't need to think it through at all. 

Instead of calling `a.Set` when the mutex is locked, try just doing `a.value = ...`